Platforms/QemuSbsaPkg/Library/QemuVirtMemInfoLib/QemuVirtMemInfoLib.c

```diff
@@ -0,0 +1,124 @@
+/** @file
+
+  Copyright (c) 2014-2017, Linaro Limited. All rights reserved.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Uefi.h>
+#include <Pi/PiMultiPhase.h>
+#include <Library/ArmLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+#include <Library/HobLib.h>
+#include <Library/MemoryAllocationLib.h>
+
+// Number of Virtual Memory Map Descriptors
+#define MAX_VIRTUAL_MEMORY_MAP_DESCRIPTORS  5
+
+//
+// mach-virt's core peripherals such as the UART, the GIC and the RTC are
+// all mapped in the 'miscellaneous device I/O' region, which we just map
+// in its entirety rather than device by device. Note that it does not
+// cover any of the NOR flash banks or PCI resource windows.
+//
+#define MACH_VIRT_PERIPH_BASE  0x08000000
+#define MACH_VIRT_PERIPH_SIZE  SIZE_128MB
+
+/**
+  Default library constructur that obtains the memory size from a PCD.

```

Comment:
```suggestion
  Default library constructor that obtains the memory size from a PCD.

```

its failing CI on this spelling